patches/net/minecraft/server/level/ServerPlayer.java.patch

```diff
@@ -91,8 +91,31 @@
          return this.isReachableBedBlock(p_9117_) || this.isReachableBedBlock(p_9117_.relative(p_9118_.getOpposite()));
      }
  
-@@ -1003,6 +_,7 @@
-                     .send(new ClientboundOpenScreenPacket(abstractcontainermenu.containerId, abstractcontainermenu.getType(), p_9033_.getDisplayName()));
+@@ -983,6 +_,11 @@
+ 
+     @Override
+     public OptionalInt openMenu(@Nullable MenuProvider p_9033_) {
++        return openMenu(p_9033_, (java.util.function.Consumer<net.minecraft.network.FriendlyByteBuf>) null);
++    }
++
++    @Override
++    public OptionalInt openMenu(@Nullable MenuProvider p_9033_, @Nullable java.util.function.Consumer<net.minecraft.network.FriendlyByteBuf> extraDataWriter) {
+         if (p_9033_ == null) {
+             return OptionalInt.empty();
+         } else {
+@@ -999,10 +_,16 @@
+ 
+                 return OptionalInt.empty();
+             } else {
+-                this.connection
+-                    .send(new ClientboundOpenScreenPacket(abstractcontainermenu.containerId, abstractcontainermenu.getType(), p_9033_.getDisplayName()));
++                if (extraDataWriter == null) {
++                    this.connection
++                        .send(new ClientboundOpenScreenPacket(abstractcontainermenu.containerId, abstractcontainermenu.getType(), p_9033_.getDisplayName()));
```

Comment:
I personally feel like that the patch is easier to read in the current state.

Do you insist on shrinking it 2 lines.

Reply:
I feel like it makes it easier to tell what is still the same as vanilla in case vanilla makes changes. You can leave it as is for now and leave this unresolved and see what other maintainers think on this comment, and then we can decide what to do and resolve it from there.